src/tools/pub/dependencies.md

```diff
@@ -31,25 +31,24 @@ dependencies:
 {% endprettify %}
 
 This YAML code creates a dependency on the `transmogrify` package
-using the default source ([pub.dev]({{site.pub}})) and
+using the default package repository ([pub.dev]({{site.pub}})) and
 allowing any version from `1.0.0` to `2.0.0` (but not including `2.0.0`).
 See the [version constraints](#version-constraints)
 section of this page for syntax details.
 
-If you want to specify a source, the syntax looks a bit different:
+If you want to specify a custom package repository,
+use the following syntax:
```

Comment:
I'm uneasy about showing the 2.15 syntax here before 2.15 has a stable channel release. One possible solution is to remove the example and do a forward link to #hosted-packages.

Reply:
Do we know which syntax people are most likely to use? Which they're most likely to have trouble with?

Reply:
Finally... Will the pre-2.15 syntax still work in 2.15+?

Reply:
> I don't want to show the 2.15 syntax here before 2.15 has a stable channel release.

Good point. I guess the first example is mainly to explain the concept of a description. We could instead let that be a git or path dependency... Or just waiting merging this until the 2.15 release.

> Do we know which syntax people are most likely to use? Which they're most likely to have trouble with?

Hopefully the new syntax! We're adding much better support for third-party repositories in 2.15, so it gets much more interesting to specify.

> Finally... Will the pre-2.15 syntax still work in 2.15+?

Yes it will.